docs/content/manual/harvester-rancher/68-Fully-airgapped-rancher-integrate-harvester-no-proxy.md

```diff
@@ -0,0 +1,511 @@
+---
+
+* Related task: [#1808](https://github.com/harvester/harvester/issues/1808) RKE2 provisioning fails when Rancher has no internet access (air-gapped)
+
+* **Note1**: in fully air gapped environment, you have to setup private docker hub registry and pull all rancher related offline image
+* **Note2**: Please use SUSE SLES JeOS image, it have `qemu-guest-agent` already installed, thus the guest VM can get IP correctly
+
+### Environment Setup
+
+Setup the airgapped harvester
+1. Fetch ipxe vagrant example with new offline feature
+https://github.com/harvester/ipxe-examples/pull/32 
+1. Edit the setting.xml file
+1. Set offline: `true`
+1. Use ipxe vagrant example to setup a 3 nodes cluster
+1. Enable vlan on `harvester-mgmt`
+1. Now harvester dashboard page will out of work
+1. Create virtual machine with name `vlan1` and id: `1`
+1. Create ubuntu cloud image from File
+
+#### Phase 1: setup airgap Harvester
+1. Set offline: true in [vagrant-pxe-harvester](https://github.com/harvester/ipxe-examples/blob/f75483563f192090dadd48051bbc3b538c30cd34/vagrant-pxe-harvester/settings.yml#L42-L44).
+2. Set 1 node harvester with sufficient resources
+3. Run ./setup_harvester.sh.
+
+#### Phase 2: Create a virtual machine for Rancher in air gapped environment 
+1. Create a virtual machine with at least 300GB storage (OS: Ubuntu Desktop 20.04)
+2. Add `harvester` and `vagrant-libvirt` network to the virtual machine  
+  - `harvester` for internal 
+  - `vagrant-libvirt` for external 
+
+#### Phase 3: Setup a private docker registry and download all the offline images for Rancher 
+1. Install VIM
+  ```
+  sudo apt update
+  sudo apt install vim
+  ```
+2. Install Docker 
+  ```
+  sudo apt-get update
+  sudo apt-get install \
+      ca-certificates \
+      curl \
+      gnupg \
+      lsb-release
+  curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+  echo \
+    "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+    $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+  sudo apt-get update
+  sudo apt-get install docker-ce docker-ce-cli containerd.io
+  ```
+3. Add current user to docker group
+  ```
+  sudo groupadd docker
+  sudo usermod -aG docker $USER
+  ```
+4. Logout and login again.
+5. Install helm
+  ```
+  curl https://baltocdn.com/helm/signing.asc | sudo apt-key add -
+  sudo apt-get install apt-transport-https --yes
+  echo "deb https://baltocdn.com/helm/stable/debian/ all main" | sudo tee /etc/apt/sources.list.d/helm-stable-debian.list
+  sudo apt-get update
+  sudo apt-get install helm
+  ```
+6. Create certs folder
+  ```
+  mkdir -p certs
+  ```
+7. Generate private registry certificate files
+  ```
+  openssl req \
+    -newkey rsa:4096 -nodes -sha256 -keyout certs/domain.key \
+    -addext "subjectAltName = DNS:myregistry.local" \
```

Comment:
It is probably worth changing the my myregistry.local to myregistry.test. On investigation it looks like `.local` commons act slightly different. it routes DNS slightly differently. However this would have to be done everywhere.

Reply:
Confirmed we can use `myregistry.test` and replace all related lines of `myregistry.local`